### PR TITLE
Android shouldn't dispatch onLayout if frame didn't change

### PR DESCRIPTION
**Pull request by hayeah**
Fixes [#7202 Android Redundant onLayout event](https://github.com/facebook/react-native/issues/7202)

### CODE REVIEW

**Comment by facebook-github-bot-4** [comment]
By analyzing the blame information on this pull request, we identified @korDen and @mkonicek to be potential reviewers.

**Comment by hayeah** [comment]
ping @mkonicek @korDen

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNode.java [hunk 1]
```diff
@@ -201,12 +201,24 @@ public void onCollectExtraUpdates(UIViewOperationQueue uiViewOperationQueue) {
...
-      mAbsoluteLeft = Math.round(absoluteX + getLayoutX());
-      mAbsoluteTop = Math.round(absoluteY + getLayoutY());
-      mAbsoluteRight = Math.round(absoluteX + getLayoutX() + getLayoutWidth());
-      mAbsoluteBottom = Math.round(absoluteY + getLayoutY() + getLayoutHeight());
+      float newLeft = Math.round(absoluteX + getLayoutX());
+      float newTop = Math.round(absoluteY + getLayoutY());
+      float newRight = Math.round(absoluteX + getLayoutX() + getLayoutWidth());
+      float newBottom = Math.round(absoluteY + getLayoutY() + getLayoutHeight());
+
+      if(newLeft == mAbsoluteLeft && newTop == mAbsoluteTop && newRight == mAbsoluteRight && newBottom == mAbsoluteBottom) {
```
Comment (mkonicek) [inline ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNode.java]: nit: space after `if`, split on multiple lines

**Comment by mkonicek** [comment]
We should probably have a dedicated API to detect screen rotation other than:

    <RootView onLayout={onDeviceRotation}>

Does this one fire if keyboard is shown for example?

**Comment by mkonicek** [comment]
Thanks for the great description in #7202.

**Comment by mkonicek** [comment]
I pinged some people internally to get more opinions.

**Comment by hayeah** [comment]
In my own code I use onLayout in a top-level component to receive the frame. Once the container component has the frame it passes the frame down the tree hierarchy, which I use as a ersatz "percentage unit". 

Something like this:

```
<Measure onLayout=(this.renderWithLayout)/>
```

In the case of keyboard event, this works exactly the same as orientation... It might be nice to have something baked-in to capture the "percentage width" concept.

File: ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNode.java (hunk 1 above)
Comment (hayeah) [inline ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactShadowNode.java]: you mean split the conditions on multiple lines?

**Comment by facebook-github-bot-0** [comment]
@hayeah updated the pull request.

**Comment by hayeah** [comment]
@mkonicek fixed nit. thanks1

**Comment by facebook-github-bot-2** [comment]
It's been a while since the last commit was reviewed and the labels show this pull request needs review. Based on the blame information for the files in this pull request we identified @korDen as a potential reviewer. Could you take a look please or cc someone with more context?

**Comment by thoblr** [comment]
Note that this PR is not only related to orientation but fixes a general difference how onLayout events are handled between iOS and Android. See my comment in the in issue #7202 for more info.

**Comment by itbrandonsilva** [comment]
Anyone know the status on this?

**Comment by rigdern** [comment]
@mkonicek any update on this PR? My team is using `v0.32.0-rc.0` and we're seeing redundant `onLayout` events on Android as well. We're now being careful when using that event because it can fire so often as to cause performance issues.

Android's `onLayout` event should only fire when the view's dimensions actually change. This is the way it behaves on iOS.

Adam Comella
Microsoft Corp.

**Comment by lacker** [comment]
Hey, sorry for letting this sit for a while. The code looks pretty straightforward to me, I think @mkonicek just wanted to double-check if this is the correct behavior. Could you fix that merge conflict?

**Comment by mkonicek** [comment]
Sorry, I got confused by #7250 mentioning device orientation. I should have read the description more carefully.

Let's ship it, I'll fix the merge conflict if there's one.

@facebook-github-bot shipit

**Comment by mkonicek** [comment]
I feel bad I haven't replied on this since May! :( We have an oncall rotation for handling pull requests now, @lacker is oncall and pinged me. Thank you so much Kevin! Hopefully this new oncall rotation will improve things :)

**Comment by mkonicek** [comment]
@hayeah Looks like this introduced another bug and was reverted, see #11096

**Comment by hayeah** [comment]
looks like it's fixed in  https://github.com/facebook/react-native/pull/11222

sorry about introducing the bug 😢

**Pull request closed without merging**